virttest/utils_test/libvirt.py

```diff
@@ -3050,29 +3051,61 @@ def hotplug_domain_vcpu(domain, count, by_virsh=True, hotplug=True):
     :param hotplug:  True means hot-plug, False means hot-unplug
     """
     if by_virsh:
-        result = virsh.setvcpus(domain, count, "--live", debug=True)
+        result = virsh.setvcpus(vm.name, count, "--live", debug=True)
     else:
-        if hotplug:
-            cpu_opt = "cpu-add"
+        result = None
+        if "ppc" in platform.machine():
+            vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm.name)
+            topology = vmxml.get_cpu_topology()
+            vcpu_count = vm.get_cpu_count()
+
+            if topology:
+                threads = int(topology["threads"])
+            else:
+                threads = 1
+            # test if count multiple of threads
+            err_str = "Expected vcpu counts to be multiples of %d" % threads
+            if hotplug:
+                err_str += ",Invalid vcpu counts for hotplug"
+            else:
+                err_str += ",Invalid vcpu counts for hotunplug"
+            if (count % threads) != 0:
+                raise exceptions.TestError(err_str)
+            if hotplug:
+                for item in range(0, int(count), threads):
+                    if item < vcpu_count:
+                        continue
+                    cmd = "device_add host-spapr-cpu-core,id=core%d,core-id=%d" % (item, item)
+                    result = virsh.qemu_monitor_command(vm.name, cmd, "--hmp",
+                                                        debug=True)
+            else:
+                for item in range(int(count), vcpu_count, threads):
+                    cmd = "device_del core%d" % item
+                    result = virsh.qemu_monitor_command(vm.name, cmd, "--hmp",
```

Comment:
```python
if ppc:
    if hotplug:
        cmd = <>
    else:
        cmd = <>
else:
    if hotplug:
        cmd = <>
    else:
        cmd = <>
return virsh.qemu_monitor_command(cmd)
```

Reply:
we could do that but problem is we need multiple `virsh.qemu_monitor_command` to be called for ppc... thats not the case with `cpu-add`

Reply:
we can form list of commands where for `cpu-add` it would be only one, also have to check the result for each iteration, current logic would return the result of last iteration it might miss if failure occurs before last iteration,
```python
for cmd in cmds:
    result = virsh.qemu_monitor_command(cmd)
    libvirt.check_exit_status(result)
return result
```

Reply:
agree on the result checks, will do check exit_status for ppc and return last...